merge sphinx branch

NarvalNudiste opened the pull request:
Bon, bah avec ça j'ai appris à utiliser vim. J'espère que ça passera sans trop de soucis 👍

Commit:
sphinx - initial commit

Signed-off-by: Guillaume Noguera <Guillaume Noguera>

Commit:
we need more titles

Commit:
test travis

Commit:
test travis

Commit:
added autodoc section + beginned apidoc, hello travis

Commit:
added autodoc section + beginned apidoc, hello travis

Commit:
fancy pictures !

Commit:
fancy pictures !

Commit:
added todo section, last push of today hopefully

Commit:
added todo section, last push of today hopefully

Commit:
Added sphinx domain section + conclusion. #hellotravis

Commit:
Added sphinx domain section + conclusion. #hellotravis

Commit:
Added fluff #hitravis

Commit:
Added fluff #hitravis

Commit:
Added conf file section, #hitravis

Commit:
Added conf file section, #hitravis

Commit:
Fixed a few typos, + added a few sources and separators #hitravis

Commit:
Fixed a few typos, + added a few sources and separators #hitravis

greut commented on source/sphinx/index.rst:
```diff
@@ -0,0 +1,323 @@
+que les fichiers conf.py et index.rst (configuration par défaut de sphinx-quickstart).
+
+
+>>>>>>>>>>>>>>>>>>>
```
:-)

greut commented:
Ajoutez des liens _intersphinx_.

NarvalNudiste commented:
Une section sur lien intersphinx, ou des liens intersphinx ?

greut commented on source/sphinx/index.rst:
```diff
@@ -0,0 +1,323 @@
+Introduction
+------------
+
+`Sphinx <http://www.sphinx-doc.org/>`_ est un générateur de documentation python libre (Licence `BSD <https://en.wikipedia.org/wiki/BSD_licenses>`_).
```
```rst
:py:mod:`sphinx`
```

greut commented on source/sphinx/index.rst:
```diff
@@ -0,0 +1,323 @@
+Il est également possible de changer le titre à afficher en haut de page au lieu du nom de fichier grâce à la notation suivante::
+	
+	..toctree::
+	:maxdepth: 1
```
avec ou sans espace?

Vous parlez de `pygments`, utilisez-le.

greut commented on source/sphinx/index.rst:
```diff
@@ -0,0 +1,323 @@
+   :align: center
+
+Néanmoins, cette méthode comporte un soucis évident : on doit quand même inclure tous les modules et classes manuellement, et ça c'est tout pourri.
+Heureusement, un utilisateur a créé un script remédiant à ce soucis : il s'agit d'apidoc. 
```
À bon?

https://he-arc.github.io/livre-python/colorama/colorama.html

NarvalNudiste replied:
Ok, alors je n'ai pas du comprendre. J'ai fait des tests en local et l'autodoc n'était affichée que pour les modules/classes apellées .. apidoc genèrait quant à lui les fichiers .rst

greut replied:
oui, avec les bons arguments tout le contenu d'un module est affiché.

greut commented on source/sphinx/index.rst:
```diff
@@ -0,0 +1,323 @@
+Le fichier de configuration
+---------------------------
+
+Le fichier conf.py, présent à la racine de notre dossier, permet de paramétrer un grand nombre de paramètres.
```
Ce fichier est plutôt auto-explicatif et surtout auto-généré, je ne suis pas sûr de voir l'intérêt de tout expliquer.

Alors que dire pour obtenir l'autodoc, il faut faire ci dans `conf.py` et ça là, etc.

NarvalNudiste replied:
Oui, j'ai pensé que c'était un peu inutile aussi, après coup

NarvalNudiste replied:
J'enlève le chapitre

Commit:
few changes + pygments higlighting #hitravis

Commit:
few changes + pygments higlighting #hitravis

NarvalNudiste commented:
Oui, non, je galère un peu pour le rebase là .. Je dois faire faux, mais il ne me signale pas de conflit

Commit:
on tente un rebase

Commit:
fixed intersphinx mapping

Commit:
Tant pis pour l'intersphinx, + il me semble bête de devoir ajouter sphinx
lui-même dans l'intersphinx_mapping .. Bref, mea culpa

Commit:
tentative sans les refs c/js, je ne comprends pas pourquoi travis
rencontre une erreur sur le type de l'argument de la fonction et non pas
sur la fonction elle même

NarvalNudiste commented:
Bon, très deçu du rendu, mais j'ai viré les réferences pour que la pr passe .. je corrigerai ça :(

greut commented on source/sphinx/index.rst:
```diff
@@ -206,22 +206,7 @@ Quelques exemples (`source <http://www.sphinx-doc.org/en/stable/domains.html>`_)
 
-> > > 
-
-.. c:function:: PyObject* PyType_GenericAlloc(PyTypeObject *type, Py_ssize_t nitems)
```
À voir, il fait des références vers ces types-là. Du coup, ils devraient être créés ou donné, peut-être.

PS: vous avez `build.sh` pour faire ça en local.

NarvalNudiste replied:
Oui, j'ai cherché où pouvaient être référencés ces types, mais je ne tombe que sur des topics assez obscurs .. Je doute pouvoir résoudre ceci en deux minutes ;_;

NarvalNudiste replied:
(J'ai un backup du fichier original, je regarderai si je peux corriger ça)

greut commented:
J'ai repris et suis en train de tenter le rebase... dur dur.
